### DISCUSSION TITLE
Align property for actors accessible in apply

### DISCUSSION THREAD

**Discussion by limbusdev** [libktx/ktx#178] [Ideas]
Hi there,
would it make sense to make Actor's align property accessible in the apply method? For some reason (difficult layout adaptions on changing size, I guess) align property has no setter, therefore this is not possible:

```kotlin
val root = Group().apply{

    x = 10f
    y = 20f
    align = Align.bottomLeft
    width = 100f
    height = 50f
}
```

It would be very pleasant to be able to do this. And code would look even tidier. Any thoughts? Cheers, Gem

**Comment by czyzby** [comment]
I don't think most actors have `align` property in the first place, so there is no way to implement this reliably. If you try using `Group.align` or `Group.setAlign`, you'll see compilation errors simply because these properties/methods do not exist.

If you're talking about actors that are stored in a table though, try the `cell` extension method:

```kotlin
table { 
  label("Test").cell(align=Align.bottom)
}
```

You can also access the cell with the lambda argument when you're using a configuration block and want to keep all of the settings next to each other:

```kotlin
table {
   label("Test") { cell ->
    this.fontScaleY = 2f // Your custom widget settings.
    cell.align(Align.bottom) // Table cell settings.
  }
}
```

Note that the `cell` extension method and the lambda parameter in the configuration block are complementary and can both work at the same time, but you usually want to keep the cell configuration in the same place.

You can find more examples in `ktx-scene2d` README. Check the sections on cell settings.
